### PR TITLE
Homework1 task1 ready

### PR DESCRIPTION
**Pull request by VytasR**
Task1 - ternary 2's calculator - ready for review

### CODE REVIEW

File: Homework1/Ternary2s/Program.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        // This method inputs integer range limits and finds integers with exactly two ternary 2's.
+        // Returns a dictionary with integer as a key and its ternary representation in string as a value.
+        static Dictionary<int, string> FindTwoTernary2s(int rangeStart, int rangeEnd)
+        {
+            var intsWithTwoTernary2s = new Dictionary<int, string>();
+
+            for (int i = rangeStart; i <= rangeEnd; i++)
+            {
+                if (AreTwoTernary2s(i))
```
Comment (radislavkireev) [inline Homework1/Ternary2s/Program.cs:58]: envelope if with {} even when one-liner inside: 

if (AreTwoTernary2s(i)) 
{
   intsWithTwoTernary2s.Add(i, ConvertDecimalToTernary(i));
}

Comment (radislavkireev) [reply Homework1/Ternary2s/Program.cs:58]: I do believe that code works, but there is a way to improve it. 

The current app converts decimal to ternary twice - partially at AreTwoTernary2s and in ConvertDecimalToTernary. 
What about convert number to ternary at first and then count 2s in it? 

string ternary = ConvertDecimalToTernary(i) 
foreach(char c in ternary) 
{
   if (c == '2')  //TBD: count "2s" and add to  intsWithTwoTernary2s
}

**Review by radislavkireev — CHANGES_REQUESTED** [review]

File: Homework1/Ternary2s/Program.cs (hunk 1 above)
Comment (VytasR) [reply Homework1/Ternary2s/Program.cs]: Hello Radislav, I updated the code. Please check. Thanks.

File: Homework1/Ternary2s/Program.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            
+            for (int i = rangeStart; i <= rangeEnd; i++)
+            {
+                string ternary = ConvertDecimalToTernary(i);
+                var sum2s = 0;
+
+                // Check for two ternary 2's
+                foreach (char c in ternary)
+                {
+                    if (c == '2') sum2s++;
```
Comment (radislavkireev) [inline Homework1/Ternary2s/Program.cs:64]: add {} even if is one-liner

File: Homework1/Ternary2s/Program.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            }
+            return intsWithTwoTernary2s;
+        }
+
+        // This method inputs integer and converts it to ternary numeral system. Returns a string.
+        static string ConvertDecimalToTernary (int number)
+        {
+            if (number == 0) return "0";
+
+            var ternaryRepresentation = new StringBuilder(5);
```
Comment (radislavkireev) [inline Homework1/Ternary2s/Program.cs:82]: why 5?

File: Homework1/Ternary2s/Program.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                    intsWithTwoTernary2s.Add(i, ternary);
+                }                
+            }
+            return intsWithTwoTernary2s;
+        }
+
+        // This method inputs integer and converts it to ternary numeral system. Returns a string.
+        static string ConvertDecimalToTernary (int number)
+        {
+            if (number == 0) return "0";
```
Comment (radislavkireev) [inline Homework1/Ternary2s/Program.cs:80]: add {} even if is one liner

File: Homework1/Ternary2s/Program.cs [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            var ternaryRepresentation = new StringBuilder(5);
+            int quotient = Math.Abs(number);
+
+            while (quotient != 0)
+            {
+                ternaryRepresentation.Insert(0, Convert.ToString(quotient % 3));
+                quotient /= 3;
+            }
+
+            if (number < 0) ternaryRepresentation.Insert(0, "-");
```
Comment (radislavkireev) [inline Homework1/Ternary2s/Program.cs:91]: add {} even if is one liner

**Review by radislavkireev — CHANGES_REQUESTED** [review]

File: Homework1/Ternary2s/Program.cs (hunk 2 above)
Comment (VytasR) [reply Homework1/Ternary2s/Program.cs]: Added in all instances

File: Homework1/Ternary2s/Program.cs (hunk 3 above)
Comment (VytasR) [reply Homework1/Ternary2s/Program.cs:92]: It covers an integer range from -80 to 242. If more characters need to be added, additional memory for the StringBuilder object will be allocated dynamically, the capacity be doubled. I believe I am saving memory too much here. What would you suggest?

Comment (radislavkireev) [reply Homework1/Ternary2s/Program.cs:92]: Use parameterless ctor, so builder can assign more memory when it needs

**Review by radislavkireev — APPROVED** [review]

**Pull request merged**